### ISSUE TITLE
Apk wrapped but not signed

### ISSUE THREAD

**Issue by AnuragM09** [msintuneappsdk/intune-app-wrapping-tool-android#33]
**Describe the bug:**
When trying to wrap the apk with  intune we are receiving a unsigned wrapped apk. Wrapping is successful but it gives a unsigned apk.
**To Reproduce**
1. Get a signed apk
2. Run the Intune wrapping command with same certs

**Expected behavior:**
Expected a wrapped and signed apk

**Screenshots and logs:**
```
Application wrapping in progress. App Wrapping Tool Version: 1.0.2676.4
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2406: error: resource 'drawable/$avd_hide_password__0' has invalid entry name '$avd_hide_password__0'. Invalid character '$avd_hide_password__0'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2407: error: resource 'drawable/$avd_hide_password__1' has invalid entry name '$avd_hide_password__1'. Invalid character '$avd_hide_password__1'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2408: error: resource 'drawable/$avd_hide_password__2' has invalid entry name '$avd_hide_password__2'. Invalid character '$avd_hide_password__2'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2409: error: resource 'drawable/$avd_show_password__0' has invalid entry name '$avd_show_password__0'. Invalid character '$avd_show_password__0'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2410: error: resource 'drawable/$avd_show_password__1' has invalid entry name '$avd_show_password__1'. Invalid character '$avd_show_password__1'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2411: error: resource 'drawable/$avd_show_password__2' has invalid entry name '$avd_show_password__2'. Invalid character '$avd_show_password__2'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2407: error: resource 'drawable/$avd_hide_password__0' has invalid entry name '$avd_hide_password__0'. Invalid character '$avd_hide_password__0'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2408: error: resource 'drawable/$avd_hide_password__1' has invalid entry name '$avd_hide_password__1'. Invalid character '$avd_hide_password__1'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2409: error: resource 'drawable/$avd_hide_password__2' has invalid entry name '$avd_hide_password__2'. Invalid character '$avd_hide_password__2'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2410: error: resource 'drawable/$avd_show_password__0' has invalid entry name '$avd_show_password__0'. Invalid character '$avd_show_password__0'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2411: error: resource 'drawable/$avd_show_password__1' has invalid entry name '$avd_show_password__1'. Invalid character '$avd_show_password__1'.
C:\Users\anuragreddy\AppData\Local\Temp\IntuneAppWrappingTool-743585394924629120\sourceApk\res\values\public.xml:2412: error: resource 'drawable/$avd_show_password__2' has invalid entry name '$avd_show_password__2'. Invalid character '$avd_show_password__2'.
The package name for this app is 'com.cbre.host.prod'. If you need to specify the package name during deployment, please use this one.
The output app was not signed because the input app was signed by Android Signature Scheme V2 or newer. This signing cannot currently be performed by the Intune App Wrapping Tool. Android requires all app packages (.apk) to be signed in order to be installed on the device. Please sign the wrapped app using the same tools used to sign the input app or using Google's apksigner. This can be done after successfully Intune App Wrapping
The application located at C:\Users\anuragreddy\Downloads\app-release-intune-unsigned.apk was successfully wrapped, but is not signed.```

**Comment by JosephNK** [comment]
Note
The Intune App Wrapping Tool does not support Google's v2 and upcoming v3 signature schemes for app signing. After you have wrapped the .apk file using the Intune App Wrapping Tool, the recommendation is to use Google's provided Apksigner tool. This will ensure that once your app gets to end user devices, it can be launched properly by Android standards.

[document link](https://docs.microsoft.com/en-us/mem/intune/developer/app-wrapper-prepare-android#security-considerations-for-running-the-app-wrapping-tool)

**Comment by meghandaly** [comment]
Duplicate of #17

**Comment by Asutosh11** [comment]
Looks like you are giving a signed APK to the App wrapping tool as Input. It expects an unsigned App. Note: even debug APKs are signed with a debug signature. 

I have written an article on checks to perform (including how to generate an unsigned APK) before starting the wrapping/signing procedure using the App wrapping tool. 

You can have a look if you want. https://ironmonkey.in/2021/05/26/prepare-your-android-app-for-wrapping-signing-using-intune-app-wrapping-tool/

**Comment by rygo-msft** [comment]
@Asutosh11 the App Wrapping Tool does support wrapping signed apks, but only using the V1 signing algorithm. The wrapping tool requires that the requested signing certificate and algorithm match those of the input apk, so the it will not sign an apk signed with the V2 or V3 signing algorithms. If you wish to wrap an apk signed using V2+ or to use a different certificate, you can do so by wrapping the apk without the signing parameters and manually signing the resulting apk using apksigner from the Android SDK.

If you are experiencing issues with the App Wrapping Tool blocking wrapping for signed apks when you have not provided signing parameters, please open an issue so that we can investigate as that is not expected to happen.
